File: backend_core_mobile/app/models/cronograma_model.py
# cronograma_model.py
import uuid
from sqlalchemy import Column, Integer, DateTime, ForeignKey, Numeric, Date, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database.session import Base

class CronogramaPago(Base):
    __tablename__ = "cr_cronograma_pagos"

    id_cuota = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    id_credito = Column(UUID(as_uuid=True), ForeignKey("cr_creditos.id_credito"), nullable=False)
    numero_cuota = Column(Integer, nullable=False)
    fecha_pago = Column(Date, nullable=False)
    monto_cuota = Column(Numeric(12, 2), nullable=False)
    capital = Column(Numeric(12, 2), nullable=False)
    interes = Column(Numeric(12, 2), nullable=False)
    saldo = Column(Numeric(12, 2), nullable=False)
    estado = Column(String(30), nullable=False) # PENDIENTE, PAGADA, VENCIDA, PARCIAL
    fecha_pago_real = Column(Date, nullable=True)
    monto_pagado = Column(Numeric(12, 2), default=0.00)
    created_at = Column(DateTime(timezone=True))

    credito = relationship("Credito", foreign_keys=[id_credito])

File: mobile_app_bcp/lib/shared/models/cronograma_model.dart
// shared/models/cronograma_model.dart
class CronogramaModel {
  final String idCuota;
  final String idCredito;
  final int numeroCuota;
  final DateTime? fechaPago;
  final double montoCuota;
  final double capital;
  final double interes;
  final double saldo;
  final String estado;
  final DateTime? fechaPagoReal;
  final double montoPagado;

  CronogramaModel({
    required this.idCuota,
    required this.idCredito,
    required this.numeroCuota,
    this.fechaPago,
    this.montoCuota = 0,
    this.capital = 0,
    this.interes = 0,
    this.saldo = 0,
    this.estado = 'PENDIENTE',
    this.fechaPagoReal,
    this.montoPagado = 0,
  });

  String get montoFormateado => 'S/ ${montoCuota.toStringAsFixed(2)}';
  String get capitalFormateado => 'S/ ${capital.toStringAsFixed(2)}';
  String get interesFormateado => 'S/ ${interes.toStringAsFixed(2)}';
  String get saldoFormateado => 'S/ ${saldo.toStringAsFixed(2)}';

  factory CronogramaModel.fromJson(Map<String, dynamic> json) {
    return CronogramaModel(
      idCuota: json['id_cuota'] ?? '',
      idCredito: json['id_credito'] ?? '',
      numeroCuota: json['numero_cuota'] ?? 0,
      fechaPago: json['fecha_pago'] != null
          ? DateTime.tryParse(json['fecha_pago'])
          : null,
      montoCuota: (json['monto_cuota'] ?? 0).toDouble(),
      capital: (json['capital'] ?? 0).toDouble(),
      interes: (json['interes'] ?? 0).toDouble(),
      saldo: (json['saldo'] ?? 0).toDouble(),
      estado: json['estado'] ?? 'PENDIENTE',
      fechaPagoReal: json['fecha_pago_real'] != null
          ? DateTime.tryParse(json['fecha_pago_real'])
          : null,
      montoPagado: (json['monto_pagado'] ?? 0).toDouble(),
    );
  }
}
